### DISCUSSION TITLE
How can I temporarily disable the command palette and its binding?

### DISCUSSION THREAD

**Discussion by and-viceversa** [Textualize/textual#5316] [Q&A] [answered]
How can I temporarily disable the command palette and its binding?

My application makes use of the command palette but there are times when it'd be nice to temporarily disable it as well.

For example, there might be a temporary Screen with a "Yes/No/Cancel" type option flow. If the command palette is accessible, the user can pivot away from that screen without choosing an available option. 

Reading through the App source, the command palette is pretty tightly coupled to the App itself. `ENABLE_COMMAND_PALETTE` is typed as a `ClassVar` so I can't modify that from within my App instance. 

Another possible method is to disable the command palette at the application level, and then reimplement the `Binding` for each Screen. That seems like either I've missed something simple or too hacky an approach.

**Comment by willmcgugan** [comment, marked as answer]
I thought this may be possible with [dynamic actions](https://textual.textualize.io/guide/actions/#dynamic-actions), but it turns out that the command palette is a special case in the footer.

Moving this to an issue, as I think it wouldn't be much effort to support.

**Comment by and-viceversa** [comment]
Issue #5321
